src/order/monotone/monovary.lean

```diff
@@ -271,3 +311,80 @@ protected lemma antivary_on_comm : antivary_on f g s ↔ antivary_on g f s :=
 ⟨antivary_on.symm, antivary_on.symm⟩
 
 end linear_order
+
+section
+variables [linear_order α] [linear_order β] (f : ι → α) (g : ι → β) {s : set ι}
+
+/-- If `f : ι → α` and `g : ι → β` are monovarying, then `monovary_order f g` is a linear order on
+`ι` that makes `f` and `g` simultaneously monotone. -/
+def monovary_order (i j : ι) : Prop :=
+prod.lex (<) (prod.lex (<) well_ordering_rel) (f i, g i, i) (f j, g j, j)
+
+instance : is_strict_total_order ι (monovary_order f g) :=
+{ trichotomous := λ i j, begin
+    convert trichotomous_of (prod.lex (<) $ prod.lex (<) well_ordering_rel) _ _,
+    { simp only [prod.ext_iff, ←and_assoc, imp_and_distrib, eq_iff_iff, iff_and_self],
+      exact ⟨congr_arg _, congr_arg _⟩ },
+    { apply_instance }
+  end,
+  irrefl := λ i, by { rw monovary_order, exact irrefl _ },
+  trans := λ i j k, by { rw monovary_order, exact trans } }
+
+variables {f g}
+
+lemma monovary_on_iff_exists_monotone_on :
+  monovary_on f g s ↔ ∃ [linear_order ι], by exactI monotone_on f s ∧ monotone_on g s :=
```

Comment:
Do square braces on exists statements actually mean anything different from no braces at all?

Reply:
Yes, they give the `linear_order ι` an autogenerated `_inst` name rather than having me name it.